### PR TITLE
#187 added landing page settings and handle redirecting on /index

### PR DESCRIPTION
**Pull request by mimmon**
We can discuss a way to handle LANDING_PAGE  - either hardcode all possibiities, or define rules to determine them dynamically. Current code suggests using default event latest or earliest, in case none is found, redirect to (empty) list of events of the same type. 
Tests TBD.

### CODE REVIEW

File: konfera/views.py [hunk 1]
```diff
@@ -1,12 +1,28 @@
...
+        timewise, event_type = LANDING_PAGE.split('_')
+        timewise = timewise.lower()
+        event_type = event_type.upper()
+    except BaseException as be:
+        raise (_('%s\nIncorrect LANDING_PAGE setting, need latest|earliest_conference|meetup', be))
+
+    events = Event.objects.published().filter(event_type=getattr(Event, event_type))
+    selected_event = getattr(events, timewise)('date_from')
+    if not selected_event:
+        messages.info(request, _('No %s event has been found. Redirected to default list.' % event_type))
```
Comment (ZoltanOnody) [inline konfera/views.py]: Does `ugettext_lazy` work in views?

File: konfera/views.py [hunk 2]
```diff
@@ -1,12 +1,28 @@
 from django.shortcuts import redirect
 from django.views.generic import ListView
+from django.shortcuts import render
```
Comment (ZoltanOnody) [inline konfera/views.py]: imports should be alphabetically

File: konfera/settings.py [hunk 3]
```diff
@@ -10,3 +10,6 @@
                             (30, _('30 min')),
                             (45, _('45 min')),
                         ))
+
+# LANDING PAGE should be something like latest_conference or earliest_meetup or similar combinations
```
Comment (mimmon) [inline konfera/settings.py]: There are four possibilities, I will specify them all.

**Comment by mimmon** [comment]
I definitely want to write to tests for it. However, it will be my first view test, so I need some time to learn how to set up and perform this type of tests.

File: konfera/views.py [hunk 4]
```diff
@@ -1,12 +1,28 @@
...
 
 def index(request):
-    latest_conference = Event.objects.published().filter(event_type=Event.CONFERENCE).latest('date_from')
-    return redirect('event_details', slug=latest_conference.slug)
+    try:
+        timewise, event_type = LANDING_PAGE.split('_')
+        timewise = timewise.lower()
+        event_type = event_type.upper()
+    except BaseException as be:
+        raise (_('%s\nIncorrect LANDING_PAGE setting, need latest|earliest_conference|meetup', be))
```
Comment (ricco386) [inline konfera/views.py]: I agree, no need for translations of error messages. This will end in log, which will be seen only by admin...

**Comment by ricco386** [comment]
@mimmon currently there is no rush for this issue, as there is already functionality for redirect (also deployed  live). So I think you can take your time and write a tests. If you are stuck and need help just ask here.

File: konfera/views.py [hunk 5]
```diff
@@ -14,12 +13,12 @@ def index(request):
...
     except BaseException as be:
-        raise (_('%s\nIncorrect LANDING_PAGE setting, need latest|earliest_conference|meetup', be))
+        raise ('%s\nIncorrect LANDING_PAGE setting, need latest|earliest_conference|meetup', be)
 
     events = Event.objects.published().filter(event_type=getattr(Event, event_type))
-    selected_event = getattr(events, timewise)('date_from')
+    selected_event = getattr(events, timewise)('date_from') if events else None
     if not selected_event:
-        messages.info(request, _('No %s event has been found. Redirected to default list.' % event_type))
+        messages.info(request, 'No %s event has been found. Redirected to default list.' % event_type)
```
Comment (ZoltanOnody) [inline konfera/views.py]: `messages` are shown to the user so they need to be translated.

File: konfera/tests/test_views.py [hunk 6]
```diff
@@ -159,3 +159,49 @@ def test_order_status_await(self):
...
+class TestIndexRedirect(TestCase):
+
+    def setUp(self):
+        self.location = Location.objects.create(
+            title='FIIT', street='Ilkovicova', city='Bratislava', postcode='841 04', state='Slovakia', capacity=400,
+        )
+
+    def test_no_conference(self):
+        self.old_meetup = Event.objects.create(
+            title='Old meetup', slug='old-meetup', description='Old meetup', event_type='meetup',
```
Comment (ZoltanOnody) [inline konfera/tests/test_views.py]: `event_type=Event.MEETUP`

same applies for the other tests

**Comment by mimmon** [comment]
I unintendedly merged this issue, I suppose everything is correct according to previous requests, however, please check it so that no issue will arise. Sorry for inconvenience.

**Pull request merged**